### PR TITLE
Create simple shaka player plugin.

### PR DESCRIPTION
**Pull request by i-zolotarenko**

### CODE REVIEW

File: packages/p2p-media-loader-shaka/src/services/engine.ts
Comment (mrlika) [inline packages/p2p-media-loader-shaka/src/services/engine.ts:1]: Test stop/start for Shaka and Hls.js instances. Add logs

**Review by mrlika — CHANGES_REQUESTED** [review]

File: packages/p2p-media-loader-shaka/src/services/manifest-parser-decorator.ts
Comment (mrlika) [inline packages/p2p-media-loader-shaka/src/services/manifest-parser-decorator.ts:1]: Test network issues

**Review by mrlika — CHANGES_REQUESTED** [review]

File: packages/p2p-media-loader-shaka/src/services/engine.ts
Comment (mrlika) [inline packages/p2p-media-loader-shaka/src/services/engine.ts:1]: Add the ability to pass Shaka module

**Review by mrlika — CHANGES_REQUESTED** [review]

File: packages/p2p-media-loader-hlsjs/src/services/segment-mananger.ts [hunk 1]
```diff
@@ -38,7 +38,8 @@ export class SegmentManager {
         }
       });
     } else if (ManifestUtil.isPlaylistManifest(manifest)) {
-      const { segments, mediaSequence } = manifest;
+      const { segments } = manifest;
+      const mediaSequence = manifest.mediaSequence || 1;
```
[the remark this answers is not in the document]
Comment (i-zolotarenko) [reply packages/p2p-media-loader-hlsjs/src/services/segment-mananger.ts:42]: I will fix it later when rewrite p2pml-hlsjs without m3u8-parser.

File: p2p-media-loader-demo/src/App.tsx [hunk 2]
```diff
@@ -53,7 +53,8 @@ function App() {
       setPlayerType("dplayer");
     }
     let player: DPlayer | Hls;
-    const url = videoUrl.advancedVideo2;
+    // const url = videoUrl.live2;
```
Comment (mrlika) [inline p2p-media-loader-demo/src/App.tsx]: Remove

File: packages/p2p-media-loader-hlsjs/src/index.ts [hunk 3]
```diff
@@ -1,2 +1,3 @@
+// eslint-disable-next-line spaced-comment
```
Comment (mrlika) [inline packages/p2p-media-loader-hlsjs/src/index.ts:1]: Remove comment

File: p2p-media-loader-demo/src/declarations.d.ts [hunk 4]
```diff
@@ -1,2 +1,2 @@
-declare module "shaka-player";
 declare module "mux.js";
+// declare module "shaka-player";
```
Comment (mrlika) [inline p2p-media-loader-demo/src/declarations.d.ts:2]: Remove comment

File: packages/p2p-media-loader-shaka/src/types/types.d.ts [hunk 5]
```diff
@@ -12,3 +12,13 @@ type HookedStream = shaka.extern.Stream & {
   streamUrl?: string;
   mediaSequenceTimeMap?: Map<number, number>;
 };
+
+export type Shaka = {
```
Comment (mrlika) [inline packages/p2p-media-loader-shaka/src/types/types.d.ts:16]: Try
```suggestion
export type Shaka = typeof window.shaka;
```

**Review by mrlika — CHANGES_REQUESTED** [review]

**Review by mrlika — APPROVED** [review]

**Pull request merged**